bindings/src/test/java/io/pravega/storage/azure/MockAzureClient.java

```diff
@@ -0,0 +1,278 @@
+
+    AzureStorageConfig config;
+
+    public MockAzureClient(AzureStorageConfig config) {
+        this.config = config;
+        if (!DATA.containsKey(config.getContainerName())) {
+            DATA.put(config.getContainerName(), new HashMap<>());
+        }
+    }
+
+    @Override
+    synchronized public AppendBlobItem create(String blobName) {
+        val objectMap = DATA.get(config.getContainerName());
+        if (objectMap == null) {
+            throw new BlobStorageException("Container doesn't exist.", null, null);
+        }
+        if (objectMap.containsKey(blobName)) {
+            throw new BlobStorageException("Container doesn't exist.", null, null);
+        }
+        objectMap.put(blobName, new InMemoryBlobData());
+        return new AppendBlobItem("", null, null, false, "",
+                Integer.toString(0), 0);
+    }
+
+    @Override
+    synchronized public boolean exists(String blobName) {
+        val objectMap = DATA.get(config.getContainerName());
+        if (objectMap == null) {
+            throw new BlobStorageException("Blob doesn't exist.", new MockHttpResponse(404, new HttpRequest(HttpMethod.HEAD, config.getEndpoint()),
+                    BlobErrorCode.BLOB_NOT_FOUND.toString()), null);
+        }
+        return objectMap.containsKey(blobName);
+    }
+
+    @Override
+    synchronized public void delete(String blobName) {
+        val objectMap = DATA.get(config.getContainerName());
+        if (objectMap == null) {
+            throw new BlobStorageException("Container doesn't exist.", new MockHttpResponse(404, new HttpRequest(HttpMethod.HEAD, config.getEndpoint()),
+                    BlobErrorCode.CONTAINER_NOT_FOUND.toString()), null);
+        }
+        if (!objectMap.containsKey(blobName)) {
+            throw new BlobStorageException("Blob doesn't exist.", new MockHttpResponse(404, new HttpRequest(HttpMethod.HEAD, config.getEndpoint()),
+                    BlobErrorCode.BLOB_NOT_FOUND.toString()), null);
+        }
+        objectMap.remove(blobName);
+    }
+
+    @Override
+    synchronized public InputStream getInputStream(String blobName, long offSetInBlob, long length) {
+        final InMemoryBlobData inMemoryBlobData = getBlobData(blobName);
+
+        // Find chunk that contains data.
+        int floorIndex = CollectionHelpers.findGreatestLowerBound(inMemoryBlobData.blockList,
+                data -> Long.compare(offSetInBlob, data.start));
+        if (inMemoryBlobData.blobLength < length || offSetInBlob > inMemoryBlobData.blobLength || inMemoryBlobData.blobLength < offSetInBlob + length) {
+            throw new IllegalArgumentException();
+        }
+        if (floorIndex == -1) {
+            throw new BlobStorageException("Container doesn't exist.", new MockHttpResponse(404, new HttpRequest(HttpMethod.HEAD, config.getEndpoint()),
+                    BlobErrorCode.CONTAINER_NOT_FOUND.toString()), null);
+        }
+
+        byte[] retValue = new byte[Math.toIntExact(length)];
```

Comment:
During development we ran into bunch of issues and had to work through all them to arrive at working code.
Please add ample comments in this code so that anyone looking at this code can understand what exactly is happening in this part.